molgenis-data-validation/src/main/java/org/molgenis/data/validation/meta/PackageValidator.java

```diff
@@ -42,4 +47,20 @@ private void validatePackageAllowed(Package aPackage) {
   private static void validatePackageName(Package aPackage) {
     NameValidator.validatePackageId(aPackage.getId());
   }
+
+  private static void validatePackageParent(Package aPackage) {
```

Comment:
I think you can use  `PackageUtils.contains()` here to check if the current package contains the package you want to set as a parent.

Reply:
Cool, wasn't aware that such a method already existed. As mentioned offline the contains algorithm prunes the children instead of the parent. Lets check outside of this PR if we can merge the algorithms.